fedimint-server/src/consensus/mod.rs

```diff
@@ -449,8 +448,12 @@ impl FedimintConsensus {
                             .await
                             .expect("Error rolling back to transaction savepoint");
                         warn!(target: LOG_CONSENSUS, %error, "Transaction failed");
-                        dbtx.insert_entry(&RejectedTransactionKey(txid), &format!("{error:?}"))
-                            .await;
+                        // do not insert a RejectedTransactionKey because there must already be
+                        // AcceptedTransactionKey
+                        if !matches!(error, TransactionReplayError(_)) {
```

Comment:
@jkitman Do we have a good way to test this? Basically `test_does_not_insert_rejected_key_on_duplicate_tx`?

Reply:
Yes, if you look at `cannot_replay_transactions` test.  We could assert the transaction status = accepted at the bottom of the test.